lib/elixir/lib/io.ex

```diff
@@ -36,7 +36,7 @@ defmodule IO do
 
   @type device :: atom | pid
   @type nodata :: {:error, term} | :eof
-  @type chardata() :: :unicode.chardata()
+  @type chardata :: binary | maybe_improper_list(char | chardata, binary | [])
```

Comment:
```suggestion
  @type chardata :: String.t() | maybe_improper_list(char | chardata, String.t() | [])
```

What do you think about using `String.t()` instead of `binary` here to represent Erlang's `unicode_binary()`?